app/gui/src/controller/searcher/component/builder.rs

```diff
@@ -164,9 +179,14 @@ impl List {
         }
     }
 
-    /// Build the list, sorting all group lists and groups' contents appropriately. (Does not sort
-    /// the [`component::List::favorites`].)
-    pub fn build(self) -> component::List {
+    /// Build the list, sorting all group lists and groups' contents appropriately. Filter the
+    /// [`component::List::favorites`] (only components with IDs passed to [`extend`] are
+    /// retained), do not sort them.
+    ///
+    /// If a [`component::Group`] in favorites is empty after the filtering, the empty group is
+    /// retained. This allows laying out the favorites in [Component
+    /// Browser](crate::controller::Searcher) in the same columns regardless of filtering.
```

Comment:
Again, docs that are not clear enough:
1. "Build the list, sorting all group lists and groups' contents appropriately." – what does "appropriately" mean?
2. "Filter the [`component::List::favorites`] (only components with IDs passed to [`extend`] are retained), do not sort them." - filter by what? Just filter them? There is no explanation of what it really does.
3. "This allows laying out the favorites in [...] in the same columns regardless of filtering" – no explanation of why this happens. Why that would not be possible if the thing would not be retained?

Also, I feel that many of docs here are in wrong place. For example, docs here describe what other algorithms do. The same applies to earlier docs, for example, the docs to fields of the List struct explain how some of the functions work. This should not be the case. Docs to a function/struct, should refer ONLY to that function or struct. If you want to describe a broader picture, use the module docs instead. In case of complex algorithms, module docs are the perfect place to explain what really happens in the file and allow function/struct docs to be shorter and more understandable, because you can explain the terms and ideas first. Please apply it to all the docs in all the PRs.